### PR TITLE
Additional Cmake support

### LINKED ISSUE #621: GaudiRun application fails if there is not input data
If there is no input data for a GaudiRun job, the data.py file is not created. Subsequently the data.py file is expected at runtime causing a failure.

### LINKED ISSUE #622: GaudiRun application prepare step
In the prepare step of the GaudiRun application, there is a "make clean" which should *not* be executed. Also the build.<arch>/ganga directory should be completely cleaned (but the directory itself should be left in place).

### LINKED ISSUE #624: GaudiRun support for different environments
I plan to add an additional `LbLogin.sh` call to the appropriate places in GaudiRun as well as exporting the CMTCONFIG to allow for different architectures of the same code to be made and submitted to different backends.

### LINKED ISSUE #626: DiracFile on Dirac backend with GaudiRun app
The output DiracFile does not work when submitting to Dirac backend.
It is fine for the Local backend and LocalFile submits ok to both Local and Dirac backend.
The error is:
```
ERROR    Error submitting job to Dirac: Traceback (most recent call last):
  File "<stdin>", line 470, in <module>                                                                                                                     
  File "<string>", line 453, in <module>                                                                                                                    
  File "/afs/cern.ch/work/m/masmith/gangadir/workspace/masmith/LocalXML/956/input/dirac-script.py", line 17, in <module>                                    
    j.setOutputData(['test_10ev.root'], outputPath='', outputSE=[])                                                                                         
TypeError: setOutputData() got an unexpected keyword argument 'outputPath'                                                                                  

ERROR   

===                                                                                                                                                         
/afs/cern.ch/work/m/masmith/gangadir/workspace/masmith/LocalXML/956/input/dirac-script.py                                                                   
===                                                                                                                                                         

ERROR   

====                                                                                                                                                        

ERROR   
# dirac job created by ganga                                                                                                                                
from DIRAC.Core.Base.Script import parseCommandLine                                                                                                         
parseCommandLine()                                                                                                                                          
from LHCbDIRAC.Interfaces.API.DiracLHCb import DiracLHCb                                                                                                    
from LHCbDIRAC.Interfaces.API.LHCbJob import LHCbJob                                                                                                        
dirac = DiracLHCb()                                                                                                                                         
j = LHCbJob()                                                                                                                                               

# default commands added by ganga                                                                                                                           
j.setName('Brunel_Nov2012_ghostFit_5_cmakeTest__{Ganga_GaudiRun_(956)}')
j.setExecutable('gaudiRun-script.py','','Ganga_GaudiRun.log')
j.setInputSandbox(['LFN:/lhcb/user/m/masmith/GangaInputFile/Job_956/__conf-3d62e484-81ea-4aeb-8d21-35fbd659f099.tgz'])
j.setOutputSandbox(['__postprocesslocations__'])
j.setInputData(['LFN:/lhcb/LHCb/Collision12/BHADRONCOMPLETEEVENT.DST/00021211/0001/00021211_00010037_1.bhadroncompleteevent.dst'])

j.setOutputData(['test_10ev.root'], outputPath='', outputSE=[])


# <-- user settings
j.setCPUTime(172800)

# user settings -->

# diracOpts added by user


# submit the job to dirac
j.setPlatform( 'ANY' )
result = dirac.submit(j)
output(result)
ERROR   
====

ERROR    BackendError: Error submitting job to Dirac: Traceback (most recent call last):
  File "<stdin>", line 470, in <module>
  File "<string>", line 453, in <module>
  File "/afs/cern.ch/work/m/masmith/gangadir/workspace/masmith/LocalXML/956/input/dirac-script.py", line 17, in <module>
    j.setOutputData(['test_10ev.root'], outputPath='', outputSE=[])
TypeError: setOutputData() got an unexpected keyword argument 'outputPath'
 (Dirac backend)
ERROR    JobManagerError: error during submit
ERROR    JobManagerError: error during submit ... reverting job 956 to the new status
```

### PR DESCRIPTION
**Pull request by rob-c**
This is a PR with some updates for GaudiRun.

I've changed the WN code to stream the stdout/stderr from the command directly as I've hit a few occasions where the stream has been lost from record.
(Plus the code feels more professional than an os.system call, although we may want to check return-codes)

I've tested and can switch between gcc48 and gcc49 with no issues on the Localhost and still execute jobs on the grid.

I'd like to get this into 6.1.21 if we've not tagged this for release yet as it puts the GaudiRun in a nice place where I only need to work on Gauss support and some general support for more complex workflows such as with unique data files per subjob as well as MassStorageFile support.

Fixes #621
Fixes #622
Fixes #624
Fixes #626

### CODE REVIEW

**Comment by egede** [comment]
> (Plus the code feels more professional than an os.system call, although we may want to check return-codes)

@rob-c Does this mean that the code doesn't check for return codes at the moment? Fine for the prototype, but if it is the case, we should file an issue against this straight away to make sure that we do not forget.

**Comment by rob-c** [comment]
@egede No the return code is returned correctly from the command line (or should be as far as I'm aware).
I'm just not 100% certain what is expected in the case of duplicating what the _GaudiPython-like_ functionality but this feels like a secondary task as I've just copied the original implementation

File: python/GangaLHCb/Lib/GaudiRunApp/GaudiRunApp.py [hunk 1]
```diff
@@ -149,6 +150,25 @@ def prepare(self, force=False):
             self.unprepare()
             raise
 
+        self.cleanGangaTargetArea(this_build_target)
```
Comment (milliams) [inline python/GangaLHCb/Lib/GaudiRunApp/GaudiRunApp.py]: It looks like `prepare()` is no longer returning a value. I guess you meant `return self.cleanGangaTargetArea(this_build_target)` or to put a `return 1` on the line after?

File: python/GangaLHCb/Lib/GaudiRunApp/GaudiRunAppHandlers.py [hunk 2]
```diff
@@ -398,6 +405,9 @@ def extractAllTarFiles(path):
                 system("tar -jxf %s" % f)
 
 def pythonScript(scriptName):
+    '''
+    This is the actual code in thr GaudiPython-like script
```
Comment (milliams) [inline python/GangaLHCb/Lib/GaudiRunApp/GaudiRunAppHandlers.py]: Typo in "thr"

File: python/GangaLHCb/Lib/GaudiRunApp/GaudiRunApp.py (hunk 1 above)
Comment (rob-c) [inline python/GangaLHCb/Lib/GaudiRunApp/GaudiRunApp.py]: Just added latter in commit

**Comment by milliams** [comment]
:+1:

**Pull request merged**